### PR TITLE
Acceptance tests: perform file actions on the files and folders that doesnot exist

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add acceptance tests for performing files action on files and folders that has been deleted from different client.

## Related Issue

- Related Issue #1720

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by kiranparajuli589** [comment]
LGTM :+1:

File: tests/acceptance/features/webUIFavorites/unfavoriteFile.feature [hunk 1]
```diff
@@ -71,3 +71,25 @@ Feature: Unmark file/folder as favorite
     Then folder "simple-folder" should not be marked as favorite on the webUI
     But folder "simple-empty-folder" should be marked as favorite on the webUI
     And folder "0" should be marked as favorite on the webUI
+
+  @issue-1720
+  Scenario: Try to favorite file and folder that does not exist
```
Comment (jasson99) [inline tests/acceptance/features/webUIFavorites/unfavoriteFile.feature]: is this favourite? Or unfavourite?

File: tests/acceptance/features/webUIFiles/deleteFilesFolders.feature [hunk 2]
```diff
@@ -266,3 +266,20 @@ Feature: deleting files and folders
...
+
+  Scenario: Try to delete file and folder that does not exist
+    Given the user has browsed to the files page
+    And the following files have been deleted by user "user1"
+      | name          |
+      | lorem.txt     |
+      | simple-folder |
+    When the user deletes file "lorem.txt" using the webUI
+    Then the error message 'Error while deleting "lorem.txt"' should be displayed on the webUI
+    And the user clears all error message from the webUI
```
Comment (jasson99) [inline tests/acceptance/features/webUIFiles/deleteFilesFolders.feature]:
```suggestion
    When the user clears all error message from the webUI
```

File: tests/acceptance/features/webUIFiles/deleteFilesFolders.feature [hunk 3]
```diff
@@ -266,3 +266,20 @@ Feature: deleting files and folders
...
+    When the user deletes folder "simple-folder" using the webUI
```
Comment (jasson99) [inline tests/acceptance/features/webUIFiles/deleteFilesFolders.feature]:
```suggestion
    And the user deletes folder "simple-folder" using the webUI
```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 4]
```diff
@@ -309,3 +309,21 @@ Feature: Sharing files and folders with internal users
...
+    Given user "user1" has logged in using the webUI
+    And the following files have been deleted by user "user1"
+      | name          |
+      | lorem.txt     |
+      | simple-folder |
+    When the user shares file "lorem.txt" with user "User Two" as "Editor" using the webUI
+    Then the error message 'Error while sharing.' should be displayed on the webUI
+    And user "UserTwo" should not be listed in the collaborators list on the webUI
+    When the user clears all error message from the webUI
+    When the user shares folder "simple-folder" with user "User Two" as "Editor" using the webUI
```
Comment (jasson99) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
    And the user shares folder "simple-folder" with user "User Two" as "Editor" using the webUI
```

File: tests/acceptance/stepDefinitions/generalContext.js [hunk 5]
```diff
@@ -33,6 +33,12 @@ Then('the error message {string} should be displayed on the webUI dialog prompt'
     .expect.element('@ocDialogPromptAlert').text.to.equal(message)
 })
 
+Then('the user clears all error message from the webUI', function () {
```
Comment (jasson99) [inline tests/acceptance/stepDefinitions/generalContext.js]:
```suggestion
When('the user clears all error message from the webUI', function () {
```

File: tests/acceptance/features/webUIFavorites/favoritesFile.feature [hunk 6]
```diff
@@ -73,3 +73,23 @@ Feature: Mark file as favorite
...
+    Then file "lorem.txt" should not be marked as favorite on the webUI
+    When the user clears all error message from the webUI
+    And the user marks folder "simple-folder" as favorite using the webUI
+    #    Then the error message 'Error while marking "simple-folder" as favorite' should be displayed on the webUI
+    Then folder "simple-folder" should not be marked as favorite on the webUI
+    When the user reloads the current page of the webUI
+    Then file "lorem.txt" should not be listed on the webUI
+    And folder "simple-folder" should not be listed on the webUI
+    And as "user1" file "lorem.txt" should not exist
+    And as "user1" folder "simple-folder" should not exist
```
Comment (skshetry) [inline tests/acceptance/features/webUIFavorites/favoritesFile.feature]: Only check via api.

Comment (skshetry) [inline tests/acceptance/features/webUIFavorites/favoritesFile.feature]: Only check via api.

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 7]
```diff
@@ -259,6 +259,19 @@ module.exports = {
...
       return this
     },
+    /**
+     *
+     * @param {string} fromName
+     */
+    downloadFile: function (fromName) {
+      this.initAjaxCounters()
+        .waitForFileVisible(fromName)
+        .useXpath()
```
Comment (skshetry) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: Isn't that redundant here? `performFileAction` already uses `.useXpath()`
````suggestion
```

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 8]
```diff
@@ -259,6 +259,19 @@ module.exports = {
 
       return this
     },
+    /**
+     *
+     * @param {string} fromName
+     */
+    downloadFile: function (fromName) {
+      this.initAjaxCounters()
```
Comment (skshetry) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: Why not just return this?

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 9]
```diff
@@ -195,7 +195,7 @@ const waitBetweenFileUploadOperations = async function () {
   timeOfLastUploadOperation = Date.now()
 }
 
-Given('the following files have been deleted by user {string}', async function (user, table) {
+Given('the following files/folders/elements have been deleted by user {string}', async function (user, table) {
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/filesContext.js]: Just to be consistent.
```suggestion
Given('the following files/folders/resources have been deleted by user {string}', async function (user, table) {
```

File: tests/acceptance/features/webUIFavorites/favoritesFile.feature [hunk 10]
```diff
@@ -73,3 +73,20 @@ Feature: Mark file as favorite
...
+  Scenario: Try to favorite file and folder that does not exist
+    Given the user has browsed to the files page
+    And the following files have been deleted by user "user1"
+      | name          |
+      | lorem.txt     |
+      | simple-folder |
+    When the user marks file "lorem.txt" as favorite using the webUI
+    #    Then the error message 'Error while marking "lorem.txt" as favorite' should be displayed on the webUI
+    Then file "lorem.txt" should not be marked as favorite on the webUI
+    When the user clears all error message from the webUI
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUIFavorites/favoritesFile.feature]: Is error message even displayed here?

Comment (dpakach) [reply tests/acceptance/features/webUIFavorites/favoritesFile.feature]: Not currently but when the issue is fixed it should be displayed.

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 11]
```diff
@@ -259,6 +259,16 @@ module.exports = {
...
       return this
     },
+    /**
+     *
+     * @param {string} fromName
+     */
+    downloadFile: function (fromName) {
+      return this.initAjaxCounters()
+        .waitForFileVisible(fromName)
+        .performFileAction(fromName, 'download')
```
Comment (skshetry) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: Should we put this into an `enum`-like? There's already 7 usages of this kind of strings.
```suggestion
        .performFileAction(fromName, FileAction.download)
```

Comment (dpakach) [reply tests/acceptance/pageObjects/FilesPageElement/filesList.js]: Moved to https://github.com/owncloud/phoenix/issues/2567

File: tests/acceptance/features/webUIFavorites/favoritesFile.feature [hunk 12]
```diff
@@ -73,3 +73,20 @@ Feature: Mark file as favorite
     And folder with path "simple-empty-folder" should be listed in the favorites page on the webUI
     And folder with path "simple-folder/simple-empty-folder" should be listed in the favorites page on the webUI
     And file with path "strängé नेपाली folder/lorem.txt" should be listed in the favorites page on the webUI
+
+  @issue-1720
+  Scenario: Try to favorite file and folder that does not exist
```
Comment (individual-it) [inline tests/acceptance/features/webUIFavorites/favoritesFile.feature]:
```suggestion
  Scenario: Try to favorite file and folder that used to exist but does not anymore
```

**Pull request merged**